Fixed FIRE, SMKE, and CO2 passing through the powder-only wall.

Sourec opened the pull request:
Caused by these gasses having a falldown value of 1. Since this may play some role in the behavior of the element (I don't know for sure, but I doubt it after some testing), I opted to change the logic guiding collisions with walls to change this behavior. Also fixed these gasses' ability to be placed inside the powder-only wall too.

jacob1 commented on src/simulation/Simulation.cpp:
```diff
@@ -2065,7 +2065,7 @@ int Simulation::eval_move(int pt, int nx, int ny, unsigned *rr)
 		if (bmap[ny/CELL][nx/CELL]==WL_ALLOWLIQUID && elements[pt].Falldown!=2)
 			return 0;
-		if (bmap[ny/CELL][nx/CELL]==WL_ALLOWSOLID && elements[pt].Falldown!=1)
+		if (bmap[ny/CELL][nx/CELL]==WL_ALLOWSOLID && (elements[pt].Falldown!=1 || pt == PT_FIRE || pt == PT_SMKE || pt || PT_CO2))
```
that ' || pt ' check looks like a mistake, you should probably fix it

jacob1 commented:
Hmm ... or instead of fixing that 'pt' thing, I think we might as well just fix the Falldown value for these elements to be correct. I've know they were wrong forever, I just haven't fixed it in case of some kind of weird compatibility thing. But I think it would make more sense just to fix it correctly since we are already breaking compatibility.

I don't think falldown being 1 affects anything except going through powder wall.

Sourec commented:
Thanks for catching that '|| pt' thing, I have no clue how that got in there. Probably leftovers from an edit or something. Anyways, I reverted the changes to the collision logic back to what they were previously and just changed the falldown values instead. I tested the changes in-game, it doesn't look like the new falldown values have any effect on behavior other than passing through walls.

jacob1 commented:
Yay fixed now
